internal/resource/config/oauth/client/oauth_client_resource.go

```diff
@@ -0,0 +1,1579 @@
+		addRequest.TokenExchangeProcessorPolicyRef = &client.ResourceLink{}
+		err := json.Unmarshal([]byte(internaljson.FromValue(plan.TokenExchangeProcessorPolicyRef, false)), addRequest.TokenExchangeProcessorPolicyRef)
+		if err != nil {
+			return err
+		}
+	}
+
+	if internaltypes.IsDefined(plan.RefreshTokenRollingGracePeriodType) {
+		addRequest.RefreshTokenRollingGracePeriodType = plan.RefreshTokenRollingGracePeriodType.ValueStringPointer()
+	}
+
+	if internaltypes.IsDefined(plan.RefreshTokenRollingGracePeriod) {
+		addRequest.RefreshTokenRollingGracePeriod = plan.RefreshTokenRollingGracePeriod.ValueInt64Pointer()
+	}
+
+	if internaltypes.IsDefined(plan.ClientSecretRetentionPeriodType) {
+		addRequest.ClientSecretRetentionPeriodType = plan.ClientSecretRetentionPeriodType.ValueStringPointer()
+	}
+
+	if internaltypes.IsDefined(plan.ClientSecretRetentionPeriod) {
+		addRequest.ClientSecretRetentionPeriod = plan.ClientSecretRetentionPeriod.ValueInt64Pointer()
+	}
+
+	if internaltypes.IsDefined(plan.TokenIntrospectionSigningAlgorithm) {
+		addRequest.TokenIntrospectionSigningAlgorithm = plan.TokenIntrospectionSigningAlgorithm.ValueStringPointer()
+	}
+
+	if internaltypes.IsDefined(plan.TokenIntrospectionEncryptionAlgorithm) {
+		addRequest.TokenIntrospectionEncryptionAlgorithm = plan.TokenIntrospectionEncryptionAlgorithm.ValueStringPointer()
+	}
+
+	if internaltypes.IsDefined(plan.TokenIntrospectionContentEncryptionAlgorithm) {
+		addRequest.TokenIntrospectionContentEncryptionAlgorithm = plan.TokenIntrospectionContentEncryptionAlgorithm.ValueStringPointer()
+	}
+
+	if internaltypes.IsDefined(plan.JwtSecuredAuthorizationResponseModeSigningAlgorithm) {
+		addRequest.JwtSecuredAuthorizationResponseModeSigningAlgorithm = plan.JwtSecuredAuthorizationResponseModeSigningAlgorithm.ValueStringPointer()
+	}
+
+	if internaltypes.IsDefined(plan.JwtSecuredAuthorizationResponseModeEncryptionAlgorithm) {
+		addRequest.JwtSecuredAuthorizationResponseModeEncryptionAlgorithm = plan.JwtSecuredAuthorizationResponseModeEncryptionAlgorithm.ValueStringPointer()
+	}
+
+	if internaltypes.IsDefined(plan.JwtSecuredAuthorizationResponseModeContentEncryptionAlgorithm) {
+		addRequest.JwtSecuredAuthorizationResponseModeContentEncryptionAlgorithm = plan.JwtSecuredAuthorizationResponseModeContentEncryptionAlgorithm.ValueStringPointer()
+	}
+
+	return nil
+
+}
+
+func (r *oauthClientResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan oauthClientResourceModel
+
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	createOauthClient := client.NewClient(plan.ClientId.ValueString(), grantTypes(plan.GrantTypes), plan.Name.ValueString())
+	err := addOptionalOauthClientFields(ctx, createOauthClient, plan)
+	if err != nil {
+		resp.Diagnostics.AddError("Failed to add optional properties to add request for OauthClient", err.Error())
```

Comment:
You could change these error messages from "OauthClient" to "OAuth Client"

Reply:
descriptions updated